meilisearch-http/src/analytics/segment_analytics.rs

```diff
@@ -210,10 +210,20 @@ impl Segment {
                     "server_provider": std::env::var("MEILI_SERVER_PROVIDER").ok(),
             })
         });
-        let infos = json!({
-            "env": opt.env.clone(),
-            "has_snapshot": opt.schedule_snapshot,
-        });
+        let infos = {
```

Comment:
Also, what I don't like here is that we serialize all of the Argument struct and pick the field that we don't want to for privacy reasons instead of doing it the other, safer, way: picking the struct fields that we care about and are safe. Doing it the other way can work with privacy if we forgot about modifying this block of code: we would lose information instead of getting too much from the user i.e. privacy argument.

Or maybe I am wrong and the #[serde(skip)] thing is enough.

Reply:
A great example is the `snapshot_dir` field that leaks information about the user filesystem that we don't want to know about, and we forgot to serde skip it and to change the value to a boolean.

Reply:
Yes, basically we need to choose if we want to maintain a big chunk of code importing every field one by one or if we want to import everything by default and *try to remember* to erase the thing containing user data.
I thought doing it like that was what @MarinPostma proposed Monday but I can change it.

Reply:
I am for skipping field as @Kerollmops said